### DISCUSSION TITLE
Restart simulation does not converge while single simulation does

### DISCUSSION THREAD

**Discussion by ZoeyChen1993** [idaholab/moose#24979] [Q&A General]
Could anyone help me with this?

I am solving a coupled thermal-chemical problem over element activating domain, the simulation converges when in a single simulation, but it doesn't converge just after couple seconds of the restart simulation. I'm not sure if this is because the mesh or the input file. For example, the simulation runs for 20s in a single simulation, if I start a simulation in 10s, then I tried to restart the simulation using a restart input file for another 10s, the restart simulation starts to not converge at 0.02s. 

The first input file starts with regular mesh creating process. There are two variables defined, then with a constant BC, an userobject block.
```
[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    ymin = 0
    xmax = 2
    ymax = 0.1
    nx = 50
    ny = 4
  []
    [initial_ink]
    input = gen
    type = SubdomainBoundingBoxGenerator
    bottom_left = '0 0 0'
    top_right = ' 0.1 0.1 0'
    block_id = 1
  []
    [sidesets]
    input = initial_ink
    type = SideSetsAroundSubdomainGenerator
    normal = '1 0 0'
    block = 1
    new_boundary = 'moving_interface'
  []
[]

[Variables]
  [temp]
    block = '1'
    order = FIRST
    family = LAGRANGE
    initial_condition = 20
  []
  [./Cure]
    order = FIRST
    family = LAGRANGE
    initial_condition = '0.1'
    block = '1'
  [../]
.........
[BCs]

  [./temp_bottom]
    type = DirichletBC
    #type = NeumannBC
    variable = temp
    boundary = bottom
    value = '200'    
  [../]
[]
..........
[UserObjects]
  [activated_elem_uo]
    type = ActivateElementsByPath
    execute_on = timestep_begin
    activate_distance = 0.06
    function_x = fx
    function_y = fy
    active_subdomain_id = 1
    expand_boundary_name = 'moving_interface'
  []
[]`
Then the restart simulation input file is only different in these blocks. All of the parameters and set up are the same.
`[Mesh]
  [gen]
    type = FileMeshGenerator
    file =2d_thermalchemo/2d_thermalchemo.e-s110
    use_for_exodus_restart = true
  []

    [sidesets]
    input = gen
    type = SideSetsAroundSubdomainGenerator
    normal = '1 0 0'
    block = 1
    new_boundary = 'moving_interface'
  []
[]

[Variables]
  [temp]
    initial_from_file_var = temp
    initial_from_file_timestep = LATEST
    block = '1'
  []
  [./Cure]
    order = FIRST
    family = LAGRANGE
    initial_from_file_var = Cure
    initial_from_file_timestep = LATEST
    block = '1'
  [../]

[]
```
The not converge result starts from here

Time Step 2, time = 0.02, dt = 0.01
 0 Nonlinear |R| = 2.795042e+01
      0 Linear |R| = 2.795042e+01
      1 Linear |R| = 8.141958e-15
Nonlinear solve did not converge due to DIVERGED_FNORM_NAN iterations 0
 Solve Did NOT Converge!
Aborting as solve did not converge

**Comment by GiudGiud** [comment]
Hello

Exodus restart is hard to set up. You need to make sure every single variable is being initialized properly.
you could output every single variable and material property to exodus on INITIAL and check that everything was initialized.
If anything is uninitialized, that's how the restart fails to converge.
One thing that is very tricky to get right are the stateful material properties.

You could try with Checkpoint restart, should be easier to set up
https://mooseframework.inl.gov/application_usage/restart_recover.html

Guillaume

**Reply by ZoeyChen1993** [reply]
Thank you very much for the solutions, I have checked the variables to exodus on INITIAL but didn't find the problem. I tried to use Checkpoint restart, it works well, the restart simulation runs normally, but it seems the out.e files can only be default in the folder with input file, where can I set additional folder to put the files.

**Reply by GiudGiud** [reply]
You can include the folder in the name of the restart file (restart_file_base iirc)
